src/CabanaPD_Boundary.hpp

```diff
@@ -168,6 +172,42 @@ struct BoundaryCondition
     }
 };
 
+template <class BCIndexSpace>
+struct BoundaryCondition<BCIndexSpace, TempBCTag>
+{
+    double _value;
+    BCIndexSpace _index_space;
+
+    BoundaryCondition( const double value, BCIndexSpace bc_index_space )
+        : _value( value )
+        , _index_space( bc_index_space )
+    {
+    }
+
+    template <class ExecSpace, class Particles>
+    void update( ExecSpace exec_space, Particles particles,
+                 RegionBoundary plane )
+    {
+        _index_space.update( exec_space, particles, plane );
+    }
+
+    template <class ExecSpace, class ParticleType>
+    void apply( ExecSpace, ParticleType& particles, double t )
+    {
+        auto temp = particles.sliceTemperature();
+        auto x = particles.sliceReferencePosition();
+        auto value = _value;
+        auto index_space = _index_space._view;
+        Kokkos::RangePolicy<ExecSpace> policy( 0, index_space.size() );
+        Kokkos::parallel_for(
+            "CabanaPD::BC::apply", policy, KOKKOS_LAMBDA( const int b ) {
+                auto pid = index_space( b );
+                // This is specifically for the thermal deformation problem
+                temp( pid ) = value * x( pid, 1 ) * t;
```

Comment:
@streeve : We need to replace Line 206 by 

temp( pid ) = value * (x( pid, 1 ) - low_corner[1]) * t;

Could you please make the change to CabanaPD_Boundary.hpp?

Reply:
done